### PR TITLE
Add common pagination component 

### PR DESCRIPTION
**Pull request by ivan-bonk**
Implement pagination functionality according to requirements
#287

### CODE REVIEW

**Comment by ivan-bonk** [comment]
Implement pagination functionality according to requirements 
https://github.com/ita-social-projects/horondi_admin/issues/287

**Review by moran711 — COMMENTED** [review]
image
Please disable possibility to type unexisting page number.

File: src/redux/selectors/contacts.selectors.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  loading: Contact.ContactLoading,
+  currentPage: Table.pagination.currentPage,
+  rowsPerPage: Table.pagination.rowsPerPage,
+  itemsCount: Table.itemsCount
+});
+
+export const selectContact = ({ Contact }) => ({
+  loading: Contact.contactsLoading,
+  contact: Contact.contact
+});
```
Comment (kapoliub) [inline src/redux/selectors/contacts.selectors.js:12]: it makes sense to move this selector to contacts.reducer

File: src/redux/selectors/material.selectors.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  loading: Material.materialLoading,
+  currentPage: Table.pagination.currentPage,
+  rowsPerPage: Table.pagination.rowsPerPage,
+  itemsCount: Table.itemsCount
+});
+
+export const selectMaterial = ({ Material }) => ({
+  loading: Material.materialLoading,
+  material: Material.material
+});
```
Comment (kapoliub) [inline src/redux/selectors/material.selectors.js:12]: same

File: src/redux/selectors/pattern.selectors.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  loading: Pattern.patternLoading,
+  currentPage: Table.pagination.currentPage,
+  rowsPerPage: Table.pagination.rowsPerPage,
+  itemsCount: Table.itemsCount
+});
+
+export const selectPattern = ({ Pattern }) => ({
+  loading: Pattern.patternLoading,
+  pattern: Pattern.pattern
+});
```
Comment (kapoliub) [inline src/redux/selectors/pattern.selectors.js:12]: same

File: src/redux/selectors/news.selectors.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  loading: News.newsLoading,
+  currentPage: Table.pagination.currentPage,
+  rowsPerPage: Table.pagination.rowsPerPage,
+  itemsCount: Table.itemsCount
+});
+
+export const selectNews = ({ News }) => ({
+  loading: News.newsLoading,
+  newsArticle: News.newsArticle
+});
```
Comment (kapoliub) [inline src/redux/selectors/news.selectors.js:12]: same

File: src/redux/table/table.reducer.js [hunk 5]
```diff
@@ -35,18 +34,28 @@ const tableState = (state = initialState, action = {}) => {
...
-      pagination: { ...state.pagination, pagesCount: action.payload }
-    };
   case SET_ITEMS_COUNT:
     return {
       ...state,
       itemsCount: action.payload
     };
   case RESET_PAGINATION:
     return initialState;
+  case UPDATE_PADAGINATION: {
```
Comment (kapoliub) [inline src/redux/table/table.reducer.js:44]: rename this case like 'UPDATE_PAGINATION'

**Comment by kapoliub** [comment]
maybe it will be good to add `enter` keypress event to page number input

File: src/redux/table/table.reducer.js (hunk 5 above)
Comment (ivan-bonk) [reply src/redux/table/table.reducer.js]: Fixed

**Comment by ivan-bonk** [comment]
> image
> Please disable possibility to type unexisting page number.

Fixed

**Comment by ivan-bonk** [comment]
> maybe it will be good to add `Enter` keypress event to page number input

Add possibility to submit with `Enter` Key

File: src/containers/table-container-generator/table-container-generator.js [hunk 6]
```diff
@@ -27,21 +32,23 @@ const TableContainerGenerator = ({ tableTitles, tableItems, pagination }) => {
           <TableBody id='table-body'>{tableItems}</TableBody>
         </Table>
       </TableContainer>
-      {pagination && <TablePaginator />}
+      {pagination && count > config.maxItemsPerPage && <TablePaginator />}
```
Comment (dieie32) [inline src/containers/table-container-generator/table-container-generator.js:35]: can it make sense to descructure this?
const { maxItemsPerPage } = config;

File: src/pages/products/product-add/product-add-submit/product-add-submit.js [hunk 7]
```diff
@@ -95,7 +95,9 @@ const ProductAddSubmit = ({
           ? productToSend[name].find((product) => product.lang === lang).value
           : productToSend[name];
 
-        return text ? <Detail key={name} title={label} text={text} /> : null;
+        return text ? (
```
Comment (dieie32) [inline src/pages/products/product-add/product-add-submit/product-add-submit.js:98]: We have lodash package for it in our project. https://lodash.com/docs/4.17.15#map

File: src/redux/contact/contact.sagas.js [hunk 8]
```diff
@@ -40,21 +41,12 @@ const {
...
-  }
-}) {
+export function* handleContactsLoad({ payload: { skip, limit } }) {
   try {
     yield put(setContactsLoading(true));
-    const contacts = yield call(getContacts, payload.skip, payload.limit);
-    yield put(
-      setContactsPagesCount(Math.ceil(contacts.count / payload.contactsPerPage))
-    );
+    console.log(skip);
```
Comment (dieie32) [inline src/redux/contact/contact.sagas.js:47]: put away console.log

Comment (ivan-bonk) [reply src/redux/contact/contact.sagas.js:47]: fixed

File: src/containers/table-container-generator/table-container-generator.js (hunk 6 above)
Comment (ivan-bonk) [reply src/containers/table-container-generator/table-container-generator.js:35]: fixed

File: src/redux/selectors/contacts.selectors.js (hunk 1 above)
Comment (ivan-bonk) [reply src/redux/selectors/contacts.selectors.js]: fixed all

**Review by kapoliub — APPROVED** [review]

File: src/pages/contacts-page/contacts-page.js [hunk 9]
```diff
@@ -62,16 +55,12 @@ const ContactsPage = () => {
...
     );
-  }, [dispatch, contactsPerPage, contactsCurrentPage]);
-
-  const changePageHandler = (e, pageIndex) => {
-    dispatch(setContactsCurrentPage(pageIndex));
-  };
+  }, [dispatch, rowsPerPage, currentPage]);
 
   const contactItems = contacts
     ? contacts.map((contact, index) => (
```
Comment (moran711) [inline src/pages/contacts-page/contacts-page.js:66]: Please change to lodash function `map` https://lodash.com/docs/4.17.15#map.

File: src/pages/material/material-page.js [hunk 10]
```diff
@@ -71,7 +52,6 @@ const MaterialPage = () => {
     openSuccessSnackbar(removeMaterial, REMOVE_MATERIAL_MESSAGE);
   };
 
-  const changeHandler = (e, value) => dispatch(setMaterialsCurrentPage(value));
   const materialItems = list.length
```
Comment (moran711) [inline src/pages/material/material-page.js:55]: same

File: src/pages/news/news-page.js [hunk 11]
```diff
@@ -57,8 +45,6 @@ const NewsPage = () => {
     openSuccessSnackbar(removeNews, REMOVE_MESSAGE);
   };
 
-  const changeHandler = (e, value) => dispatch(setCurrentPage(value));
-
   const newsItems =
```
Comment (moran711) [inline src/pages/news/news-page.js:48]: same

File: src/pages/pattern/pattern-page.js [hunk 12]
```diff
@@ -65,8 +51,6 @@ const PatternPage = () => {
     openSuccessSnackbar(removePattern, PATTERN_REMOVE_MESSAGE);
   };
 
-  const changeHandler = (e, value) => dispatch(setPatternsCurrentPage(value));
-
   const patternItems =
     list !== undefined
```
Comment (moran711) [inline src/pages/pattern/pattern-page.js:55]: same

File: src/pages/users/components/user-tab/user-tab.js [hunk 13]
```diff
@@ -9,14 +9,15 @@ import { formatPhoneNumber } from '../../../../utils/format-phone-number';
...
 const tableTitles = config.tableHeadRowTitles.users.userTab;
 const { unknownUser } = config.labels.user;
 
 const UserTab = (props) => {
   const { list, onDelete } = props;
-  const userLoading = useSelector(({ Users }) => Users.userLoading);
+  const { userLoading, itemsCount } = useSelector(selectUserLoadAndItemsCount);
   const dispatch = useDispatch();
 
   const usersItems = list.map((userItem) => (
```
Comment (moran711) [inline src/pages/users/components/user-tab/user-tab.js:23]: same

**Review by dieie32 — APPROVED** [review]

File: src/pages/contacts-page/contacts-page.js (hunk 9 above)
Comment (ivan-bonk) [reply src/pages/contacts-page/contacts-page.js]: Fixed all

**Review by moran711 — APPROVED** [review]

**Pull request merged**